File: script/create_hit.rb
require 'json'
require 'mturk'

# Read from config file
file = File.read("config.json")
config = JSON.parse(file)["hitCreation"]

# For production:
# mturk = Amazon::WebServices::MechanicalTurkRequester.new :Host => :Production

# For testing:
mturk = Amazon::WebServices::MechanicalTurkRequester.new :Host => :Sandbox

QUALIFICATION_REQS = [
	{
		:QualificationTypeId => Amazon::WebServices::MechanicalTurkRequester::LOCALE_QUALIFICATION_TYPE_ID,
		:Comparator => 'EqualTo',
		:LocaleValue => {:Country => 'US'}
	},
	{
		:QualificationTypeId => Amazon::WebServices::MechanicalTurkRequester::TOTAL_NUMBER_OF_HITS_APPROVED_QUALIFICATION_TYPE_ID,
		:Comparator => 'GreaterThanOrEqualTo',
		:IntegerValue => 0
	},
	{
		:QualificationTypeId => Amazon::WebServices::MechanicalTurkRequester::APPROVAL_RATE_QUALIFICATION_TYPE_ID,
		:Comparator => 'GreaterThanOrEqualTo',
		:IntegerValue => 95
	}
]

for i in 1..config["numTasks"]
    questionText = "<ExternalQuestion xmlns=\"http://mechanicalturk.amazonaws.com/AWSMechanicalTurkDataSchemas/2006-07-14/ExternalQuestion.xsd\">\n"
    questionText += "<ExternalURL>" + config["taskUrl"]
    questionText += "</ExternalURL>\n  <FrameHeight>700</FrameHeight>\n</ExternalQuestion>"

    mturk.createHIT(
    	:Title => config["title"],
    	:Description => config["description"],
    	:MaxAssignments => config["numAssignments"],
    	:Reward => { :Amount => config["rewardAmount"], :CurrencyCode => "USD" },
    	:AssignmentDurationInSeconds => config["duration"],
    	:LifetimeInSeconds => config["lifetime"],
    	:Question => questionText,
    	:Keywords => config["keywords"],
    	:QualificationRequirement => QUALIFICATION_REQS
    )

    puts "Created task #{i}"
end

File: script/create_hit.py
from boto3 import client
import json

with open('../config.json', 'r') as f:
    config = json.loads(f.read())['hitCreation']

if config['production']:
    endpoint_url = 'https://mturk-requester.us-east-1.amazonaws.com'
else:
    endpoint_url = 'https://mturk-requester-sandbox.us-east-1.amazonaws.com'

cl = client('mturk', region_name='us-east-1', endpoint_url=endpoint_url)


def create_hit():
    quals = [
       {
           'QualificationTypeId': '00000000000000000071',
           'Comparator': 'EqualTo',
           'LocaleValues': [{
               'Country': 'US',
           }],
       },
       {
           'QualificationTypeId': '000000000000000000L0',
           'Comparator': 'GreaterThanOrEqualTo',
           'IntegerValues': [
               95
           ],
       },
    ]

    questionText = "<ExternalQuestion xmlns=\"http://mechanicalturk.amazonaws.com/AWSMechanicalTurkDataSchemas/"
    questionText += "2006-07-14/ExternalQuestion.xsd\">\n<ExternalURL>" + config['taskUrl']
    questionText += "</ExternalURL>\n  <FrameHeight>700</FrameHeight>\n</ExternalQuestion>"

    cl.create_hit(
        MaxAssignments=config['numAssignments'],
        AutoApprovalDelayInSeconds=604800,
        LifetimeInSeconds=config['lifetime'],
        AssignmentDurationInSeconds=config['duration'],
        Reward=config['rewardAmount'],
        Title=config['title'],
        Keywords=config['keywords'],
        Description=config['description'],
        Question=questionText,
        QualificationRequirements=quals,
        )


for i in range(config['numTasks']):
    create_hit()
